File: First Bad Version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        long long front = 0, back = n;
        
        while(front <= back){
            long long mid = (front + back) / 2;
            
            if(!isBadVersion(mid)){
                front = mid + 1;
            }
            else{
                back = mid - 1;
            }
        }
        
        return front;
    }
};

File: First Bad Version.js
var solution = function(isBadVersion) {
    return function(n) {
        let front = 0;
        
        while(front <= n){
            const mid = Math.floor((front + n) / 2);
            
            if(!isBadVersion(mid)){
                front = mid + 1;
            } else{
                n = mid - 1;
            }
        }
        
        return front;
    };
};